### PR TITLE
Cache downloaded deps in Travis

### PR DESCRIPTION
**Pull request by DaoWen**

### CODE REVIEW

**Comment by DaoWen** [comment]
This shaves at least a couple of minutes off of each of our tests:

| Test | No Caching [[link](https://travis-ci.org/twosigma/Cook/builds/302028803)] | With Caching [[link](https://travis-ci.org/twosigma/Cook/builds/302317504)] | Delta |
|---|---|---|---|
| Executor | 4m46s | 2m52s | 1m54s |
| Scheduler | 6m53s | 4m28s | 2m25s |
| Job Client | 3m32s | 1m02s | 2m30s |
| Simulator | 9m16s | 6m03s | 3m13s |
| Integration (Cook Executor) | 13m14s | 10m13s | 3m01s |
| Integration (Mesos Executor) | 12m46s | 10m55s | 1m51s |
| Scheduler Benchmark | 12m06s | 10m19s | 1m47s |
| Integration Explicit | 9m22s | 6m17s | 3m05s |
| Total | 71m55s | 52m09s | 19m46s | 19m46s |

**Comment by shamsimam** [comment]
Any way to cache the docker images?

**Comment by DaoWen** [comment]
@shams I think there are [ways to do that](https://stackoverflow.com/questions/35305492/cache-docker-images-on-travis-ci), but that would require more work than I had to do for this patch. This one was pretty straightforward. It might make more sense to just remove some of our docker dependencies rather than trying to cache the images. But I'll definitely look into it when I have some spare time.

File: .travis.yml [hunk 1]
```diff
@@ -8,17 +8,23 @@ before_install:
   - echo "deb https://repos.mesosphere.io/ubuntu/ trusty main" | sudo tee /etc/apt/sources.list.d/mesosphere.list
   - sudo apt-get update -qq
   - sudo apt-get install mesos -y </dev/null
-  - pyenv install 3.6.2
+  - eval "$(pyenv init -)"
+  - pyenv install --skip-existing 3.6.2
```
Comment (dposada) [inline .travis.yml]: Is it possible to add comments in this file? If so, it might be nice to explain why we're doing some of this. If not, no big deal

**Comment by DaoWen** [comment]
@pschorf @dposada I think this is ready to merge.

**Pull request merged**